lightning/src/offers/static_invoice.rs

```diff
@@ -312,6 +313,12 @@ impl StaticInvoice {
 	pub fn signature(&self) -> Signature {
 		self.signature
 	}
+
+	pub(crate) fn from_same_offer(&self, invreq: &InvoiceRequest) -> bool {
+		let invoice_offer_tlv_stream = TlvStream::new(&self.bytes).range(OFFER_TYPES);
+		let invreq_offer_tlv_stream = TlvStream::new(invreq.bytes()).range(OFFER_TYPES);
+		invoice_offer_tlv_stream.eq(invreq_offer_tlv_stream)
+	}
```

Comment:
We may want to compare `OfferId`s instead. It's a little more work to compute, but it would handle the experimental range once #3237 lands.

Reply:
Check out the latest push. We may want to also get rid of the `from_valid_offer_tlv_stream` method since it's a bit redundant now.

Reply:
We could but it is fragile to repeat it . We could re-factor it to re-use the same code, though in #3237.

Reply:
Sounds good, went ahead and removed the relevant commits.